demo/app/components/select/select.component.ts

```diff
@@ -1,80 +1,398 @@
-import { Component } from '@angular/core';
+import {
+  ChangeDetectorRef,
+  Component,
+  OnInit,
+} from '@angular/core';
 import {
   FormBuilder,
   FormGroup,
   Validators,
+  FormControl,
 } from '@angular/forms';
 import {
+  BehaviorSubject,
   Observable,
   of,
 } from 'rxjs';
+import {
+  map,
+  startWith,
+} from 'rxjs/operators';
+
+
+
+export interface State {
+  name: string;
+  population: string;
+}
+
 
 
 @Component({
   selector: 'demo-select',
   templateUrl: './select.component.html',
 })
-export class SelectComponent {
+export class SelectComponent implements OnInit {
   simpleItems: number[] = [1, 2, 3, 4];
   initialSimpleItemsSelection = 2;
-  items: Observable<any[]> = of([
+  obj = {
+    id: 3,
+    text: 'baz',
+  };
+  singleWithCustomTrigger: Observable<any[]> = of([
     {
-      foo: 'Foo',
+      foo: 'Foo bar baz foo bar baz foo bar baz',
       slug: 'foo',
+      bar: {
+        id: 1,
+        text: 'foo',
+      },
     },
     {
-      foo: 'Bar',
+      foo: 'Bar Hic saepe ad sunt temporibus.',
       slug: 'bar',
+      bar: {
+        id: 2,
+        text: 'bar',
+      },
     },
     {
-      foo: 'Baz',
-      slug: 'baz',
+      // tslint:disable: max-line-length
+      foo: 'Consequuntur eum eveniet accusamus ea saepe. Alias occaecati eos reprehenderit expedita. Ab perferendis nemo molestias nulla est inventore voluptate.',
+      // tslint:enable: max-line-length
+      /*
+       *foo: 'Consequuntur eum eveniet accusamus ea saepe.',
```

Comment:
vestigial comment?

Reply:
These are all just left for easy demo changes. Rather than creating a ton of specific demos, we can just go in here and change what is passed in or what the default value is etc without having to rewrite the fake code all over again.